rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/rmnode/RMNodeImpl.java

```diff
@@ -492,4 +493,23 @@ public boolean isDeploying() {
         return false;
     }
 
+    @Override
+    public void addToken(String token) {
+        PrincipalPermission principalPermission = (PrincipalPermission) getUserPermission();
+        principalPermission.addPermission(new TokenPrincipal(token));
```

Comment:
I don't understand where the local variable `principalPermission` is used ?

Reply:
I add to the `principalPermission` new permission